src/OSPSuite.Presentation/Presenters/ObservedData/DataRepositoryDataPresenter.cs

```diff
@@ -5,197 +5,50 @@
 using OSPSuite.Assets;
 using OSPSuite.Core.Commands;
 using OSPSuite.Core.Domain;
-using OSPSuite.Core.Domain.Data;
 using OSPSuite.Core.Domain.Services;
 using OSPSuite.Core.Domain.UnitSystem;
 using OSPSuite.Core.Events;
-using OSPSuite.Core.Extensions;
 using OSPSuite.Presentation.DTO;
 using OSPSuite.Presentation.Views.ObservedData;
 using OSPSuite.Utility.Events;
-using OSPSuite.Utility.Exceptions;
 using OSPSuite.Utility.Extensions;
-using DataColumn = System.Data.DataColumn;
+using DataColumn = OSPSuite.Core.Domain.Data.DataColumn;
 
 namespace OSPSuite.Presentation.Presenters.ObservedData
 {
    public interface IDataRepositoryDataPresenter : IBaseDataRepositoryDataPresenter<IDataRepositoryDataView>, IDataRepositoryItemPresenter,
       IUnitsInColumnPresenter<int>,
       IListener<ObservedDataValueChangedEvent>,
       IListener<ObservedDataTableChangedEvent>
-   {
-      void ValueIsSet(CellValueChangedDTO cellValueChangedDTO);
-      void RemoveData(int rowIndex);
-      void AddRow();
-      void AddData(int rowIndex);
-      IEnumerable<string> GetCellValidationErrorMessages(int rowIndex, int columnIndex, string newValue);
+   { 
       int NumberOfObservations { get; }
-      void DisableEdition();
    }
 
-   public class DataRepositoryDataPresenter : BaseDataRepositoryDataPresenter<IDataRepositoryDataView, IDataRepositoryDataPresenter>, IDataRepositoryDataPresenter
+   public class DataRepositoryDataPresenter : BaseDataRepositoryDataPresenter<IDataRepositoryDataView, IDataRepositoryDataPresenter>,
+      IDataRepositoryDataPresenter
    {
       private readonly IDataRepositoryExportTask _dataRepositoryTask;
       private readonly IEditObservedDataTask _editObservedDataTask;
 
-      public DataRepositoryDataPresenter(IDataRepositoryDataView view, IDataRepositoryExportTask dataRepositoryTask, IEditObservedDataTask editObservedDataTask)
+      public DataRepositoryDataPresenter(IDataRepositoryDataView view, IDataRepositoryExportTask dataRepositoryTask,
+         IEditObservedDataTask editObservedDataTask)
          : base(view)
       {
          _dataRepositoryTask = dataRepositoryTask;
          _editObservedDataTask = editObservedDataTask;
       }
 
-      public void DisableEdition()
-      {
-         View.DisableEdition();
-      }
-
       protected override DataTable MapDataTableFromColumns()
       {
          return _dataRepositoryTask.ToDataTable(_observedData, new DataColumnExportOptions { ForceColumnTypeAsObject = true }).First();
       }
 
-      public void ValueIsSet(CellValueChangedDTO cellValueChangedDTO)
-      {
-         this.DoWithinLatch(() => AddCommand(_editObservedDataTask.SetValue(_observedData, mapFrom(cellValueChangedDTO))));
-      }
-
-      public void RemoveData(int rowIndex)
-      {
-         if (_observedData.BaseGrid.Count > rowIndex)
-            AddCommand(_editObservedDataTask.RemoveValue(_observedData, rowIndex));
```

Comment:
Now I think that all the Remove Value etc.. code neecs also to be removed (and the associated command) at the domain level. This is not used anymore. 

  /// <summary>
      /// Command which modifies the units on a specific column of a data repository
      /// </summary>
      /// <param name="dataRepository">The data repository to be modified</param>
      /// <param name="columnId">The column id of the column being modified in the data repository</param>
      /// <param name="newUnit">The new unit</param>
      /// <returns>The command which was used to modify the repositories</returns>
      ICommand SetUnit(DataRepository dataRepository, string columnId, Unit newUnit);

      /// <summary>
      /// Command which removes values from the data repository
      /// </summary>
      /// <param name="observedData">The data repository being modified</param>
      /// <param name="dataRowIndex">The index of the row being removed</param>
      /// <returns>The command which was used to modify the repositories</returns>
      ICommand RemoveValue(DataRepository observedData, int dataRowIndex);

      /// <summary>
      /// Command which adds values to a data repository
      /// </summary>
      /// <param name="observedData">The repository being modified</param>
      /// <param name="dataRowAdded">An entity containing new values for each column of the data repository</param>
      /// <returns>The command which was used to modify the repositories</returns>
      ICommand AddValue(DataRepository observedData, DataRowData dataRowAdded);
/// <summary>
      /// Command which modifies the value of an observed data point in a data repository
      /// </summary>
      /// <param name="observedData">The repository being modified by the command</param>
      /// <param name="cellValueChanged">An entity indicating the new and old cell values as well as the row and column where the data is found</param>
      /// <returns>The command used to modify the repository</returns>
      ICommand SetValue(DataRepository observedData, CellValueChanged cellValueChanged);


And  then the Command, the Parameter CellValueChanged etc..

Reply:
and I would rename IEditObservedDataTask to IEditObservedDataMetaDataTask or better `IObservedDataMetaDataTask`
as we only have methods related to MetaData

Reply:
commands removed and interface renamed